packages/core/src/components/editable-text/editableText.tsx

```diff
@@ -22,6 +22,14 @@ import { DISPLAYNAME_PREFIX, type IntentProps, type Props } from "../../common/p
 import { clamp } from "../../common/utils";
 
 export interface EditableTextProps extends IntentProps, Props {
+    /**
+     * Custom attributes that will be passed to the underlying textarea element when
+     * `multiline={true}`. This allows you to specify additional attributes like
+     * `aria-` attributes, `data-` attributes, etc.
+     */
+    customTextareaAttributes?: {
```

Comment:
I think it's plausible that a consumer of this component might want to add custom attributes even when the component isn't multiline. Could we extend this prop to cover both cases? Something like:
```suggestion
    customInputAttributes?: {
```
```tsx
return multiline ? (
    <textarea ref={this.refHandlers.input} {...props} {...customInputAttributes} />
) : (
    <input ref={this.refHandlers.input} type={type} {...props} {...customInputAttributes} />
);
```

We should also consider limiting the scope of attributes we can pass to just those of textarea and input elements:
```ts
type HTMLInputProps = React.InputHTMLAttributes<HTMLInputElement>;
type HTMLTextAreaProps = React.TextareaHTMLAttributes<HTMLTextAreaElement>;

{ customInputAttributes?: HTMLInputProps & HTMLTextAreaProps; }
```

Reply:
Makes perfect sense! I should have updated the tests and the prop to allow for this. Thanks for reviewing!